### PR TITLE
If the date in a tag is invalid, skip it

### LINKED ISSUE #1506: ValueError from bad audio tags
audio/tags.py:

> def convert_taglist(taglist):
> ...
> date = datetime.date(
>                     value.get_year(), value.get_month(), value.get_day())

This raises a ValueError that is not catched when audio track has badly set year. 
I suggest that if the year exceeds datetime.MAXYEAR then use datetime.MAXYEAR or 0 as the audio itself is probably not broken.

### PR DESCRIPTION
**Pull request by palfrey**
Fixes #1506

### CODE REVIEW

File: mopidy/audio/tags.py [hunk 1]
```diff
@@ -44,10 +44,16 @@ def convert_taglist(taglist):
...
-                    value.get_year(), value.get_month(), value.get_day())
-                result[tag].append(date.isoformat().decode('utf-8'))
-            if isinstance(value, Gst.DateTime):
+                try:
+                    date = datetime.date(
+                        value.get_year(), value.get_month(), value.get_day())
+                    result[tag].append(date.isoformat().decode('utf-8'))
+                except ValueError:
+                    logger.log(
+                        logging.DEBUG,
```
Comment (adamcik) [inline mopidy/audio/tags.py]: Should just be `logger.debug(...)` no need to use `log()`.

**Comment by adamcik** [comment]
Note to self, also credit @mlang for #1517 even though this change will go in instead.

File: mopidy/audio/tags.py (hunk 1 above)
Comment (palfrey) [inline mopidy/audio/tags.py]: Fixed

**Comment by adamcik** [comment]
Thanks for the contribution, this has been backported to 2.0.x so we can hopefully do a point release with this really soon.

**Pull request merged**